core/src/test/java/com/graphhopper/storage/GraphHopperStorageCHTest.java

```diff
@@ -61,8 +61,10 @@ private GraphHopperStorage newGHStorage(boolean is3D, boolean forEdgeBasedTraver
     private GraphHopperStorage newGHStorage(Directory dir, boolean is3D, boolean forEdgeBasedTraversal, int segmentSize) {
         GraphHopperStorage graph = new GraphBuilder(encodingManager)
                 .setDir(dir).set3D(is3D).withTurnCosts(true).setSegmentSize(segmentSize).build();
-        FastestWeighting weighting = new FastestWeighting(carEncoder, new DefaultTurnCostProvider(carEncoder, graph.getTurnCostStorage()));
-        graph.addCHGraph(new CHProfile(weighting, forEdgeBasedTraversal));
+        for (FlagEncoder encoder : encodingManager.fetchEdgeEncoders()) {
+            FastestWeighting weighting = new FastestWeighting(encoder, new DefaultTurnCostProvider(encoder, graph.getTurnCostStorage()));
+            graph.addCHGraph(new CHProfile(weighting, forEdgeBasedTraversal));
+        }
```

Comment:
@easbar can you have a look into this test. I'm getting:

```
java.lang.IllegalStateException: Configured CH profile: fastest|car|edge_based=false is not contained in loaded weightings for CH[fastest|car|edge_based=false|u_turn_costs=-1, fastest|bike|edge_based=false|u_turn_costs=-1].
You configured: [fastest|car|edge_based=false, fastest|bike|edge_based=false]
```

Is this due to the compatibility layer? Should we remove this :) ?

And I also noticed that Objects.hash is used in CHProfile which is not available in Java 7 but replacing this should be simple.

Reply:
Ah, thanks a lot! This way I learn the new stuff pretty fast :)

Reply:
> And cannot make it working. IMO both profiles are identical.

I think I fixed this with the last commit. The problem was just an inconsistent way of how the weightings for CH profiles were created for the graphs. 

> Is this due to the compatibility layer? Should we remove this :) ?

No this was not because of the compatibility layer. The problem was rather that in one case the weighting had a no turn cost provider and in the other a default turn cost provider.

Reply:
This error would have been caught by this check: https://github.com/graphhopper/graphhopper/pull/1863#discussion_r371029426 (had we not removed it...) 

But if we really want to improve something here I think we should refactor these graph storage tests, I am often spending time figuring out which of the createGHStorage methods is actually being called, because they are overriding and calling each other, which can be very confusing.

Reply:
> But if we really want to improve something here 

Yeah, I noticed this too. The number of methods is just too much there.

> This error would have been caught by this check: #1863 (comment) (had we not removed it...)

Haha :) ... maybe we reintroduce it when time shows this is ugly.